app/elements/coder/code-editor/file-explorer/explorer-folder.html

```diff
@@ -0,0 +1,182 @@
+<link rel="import" href="/bower_components/iron-flex-layout/iron-flex-layout-classes.html">
+<link rel="import" href="/bower_components/iron-icon/iron-icon.html">
+<link rel="import" href="/bower_components/paper-toast/paper-toast.html">
+
+<link rel="import" href="explorer-file.html">
+
+<dom-module id="explorer-folder">
+	<template>
+		<style include="iron-flex">
+			.folder-wrapper:hover {
+				background-color: lightgray;
+				cursor: pointer;
+			}
+			.folder-wrapper {
+				@apply(--layout-horizontal);
+				padding: 3px;
+			}
+			#folder-name {
+				margin-left: 2px;
+			}
+			paper-input {
+				height: 30px;
+			}
+			input {
+				width: calc(100% - 26px);
+			}
+		</style>
+		<div id="wrapper" class="folder-wrapper">
+			<iron-icon icon="folder" ></iron-icon>
+			<template is="dom-if" if="{{editMode}}">
+				<input type="text" value="{{newName::input}}" on-keyup="_confirmRename" autofocus>
+			</template>
+			<template is="dom-if" if="{{!editMode}}">
+				<div id="folder-name">{{meta.name}}</div>
+			</template>
+		</div>
+		<template is="dom-repeat" items="{{meta.children}}">
+			<template is="dom-if" if="{{item.isFolder}}">
+				<explorer-folder meta="{{item}}" on-rename="_checkRename" on-delete="_deleteChild"></explorer-folder>
+			</template>
+			<template is="dom-if" if="{{!item.isFolder}}">
+				<explorer-file meta="{{item}}" on-rename="_checkRename" on-delete="_deleteChild"></explorer-file>
+			</template>
+		</template>
+		<paper-toast id="toast"></paper-toast>
+	</template>
+	<script type="application/javascript" src="/scripts/behaviors/file-explorer-item-behavior.js"></script>
+	<script>
+		Polymer({
+			is: 'explorer-folder',
+			behaviors: [
+				Behaviors.FileExplorerItemBehavior
+			],
+			addFolder: function(folderName){
```

Comment:
you can add `editMode = true` by default so the user does not rightclick and rename.